src/analysis/type_utils.ml

```diff
@@ -120,7 +120,13 @@ module Printtyp = struct
 
   let modtype env ppf mty =
     let smart ppf = function 
-      | Types.Mty_ident _ | Mty_alias _ -> verbose_modtype env ppf mty
+      | Types.Mty_ident _ | Mty_alias _ -> 
+        Format.pp_print_string ppf "(* Module type: *)\n";
+        modtype ppf mty;
+        Format.pp_print_newline ppf ();
+        Format.pp_print_newline ppf ();
+        Format.pp_print_string ppf "(* Inside module type: *)\n";
+        verbose_modtype env ppf mty
```

Comment:
I really dislike that, when Merlin is asked for a type, the `type` field of the answer should be  a type, not some formatted text with types in it.

An equation would be better: `(module M) = sig ...end` but even that seems a bit ankward.

The users asked for the type of a module, she doesn't need it to be repeated in the answer. IMO we should just drop that commit.